### ISSUE TITLE
VSCode RequiredArgumentMissing when installing C++ extension (fix)

### ISSUE THREAD

**Issue by tgarc** [fjacomme/bela#1]
First, just want to thank you for providing these instructions, this setup is pretty dope!

Second, I personally encountered a "RequiredArgumentMissing" error from vscode when trying to install C/C++ extension on the Bela, so I just wanted to provide a fix (hack) here in case anyone else encountered it:

1. Download extension from marketplace (ARM32 architecture). 
Note: There is a "Download Extension" link on the Extension's page

2. Copy extension file (.vsix) to bela e.g. :
`scp ms-vscode.cpptools-1.14.1\@linux-armhf.vsix root@bela.local:`

3. From bela, install extension with:
`code --install-extension ~/ms-vscode.cpptools-1.14.1\@linux-armhf.vsix`

**Comment by D300** [comment]
Hey, do i understand right that you are installing the C/C++ extensions ON the Bela? 
that means you have also vscode on the bela right? ...

probably that is meant by the instruction:    "Install Microsort C/C++ extension in SSH: bela.local" 

I´m having the issue that the remote filesystem is not detected by vscode and I cant build 
https://forum.bela.io/d/1066-bela-projects-in-vscode-with-syntax-checking/3

**Comment by tgarc** [comment]
That's correct, the extension is installed on the bela. When the ssh remote extension connects to bela it sets up a vscode server automatically - there's no need to manually install vscode on the bela.

You mentioned using the instructions here:

https://learn.bela.io/using-bela/technical-explainers/scripts/#build_projectsh

That method is completely orthogonal to the method described in this repo. If what you want to accomplish is to set up the dev environment described by this repo, ignore those instructions and follow those detailed in [vscode.md](https://github.com/fjacomme/bela/blob/master/vscode.md#how-to) instead.

**Comment by D300** [comment]
thanks a lot for guiding me in the right direction!

as hint: bela.local wants a password. 
due to this link  ( https://forum.bela.io/d/2309-shh-password )
i used [192.168.6.2]

this also helped me: https://code.visualstudio.com/docs/remote/ssh

and a question: how long did it take for you to download the c/c++ extension?

do you have disconnecting and reconnecting issues? how is bela connected to your computer usb and/or ethernet?

right now im wondering how to add the .clang-format file to the root of the remote machine.
Is it possible to use the extension named "CLang Format" file from Xaver Hellauer?

**Comment by tgarc** [comment]
A lot of questions! It did take a long time to download, and the connection is a little iffy at times, but generally it works. You have to take into account that vscode launches some processes on the bela which can affect the responsiveness. Can't really help with the other questions.

Anyway this is probably not the best place for this discussion, let's move it over to the bela forum where more people will see it.
